perspective.el

```diff
@@ -1166,13 +1166,20 @@ visible in a perspective as windows, they will be saved as
 '*scratch* (persp)' buffers."
   (interactive (list
                 (read-file-name "Save perspective state to file: "
-                                persp-state-default-file
-                                persp-state-default-file)
-                t))
+                                (if (and
+                                     (bound-and-true-p persp-state-default-file)
+                                     (file-directory-p (file-name-directory persp-state-default-file)))
+                                    (file-name-directory persp-state-default-file)
+                                  "")
+                                (if (and
+                                     (bound-and-true-p persp-state-default-file)
+                                     (file-readable-p persp-state-default-file))
+                                    (file-name-nondirectory persp-state-default-file)
+                                  ""))))
   (unless persp-mode
     (message "persp-mode not enabled, nothing to save")
     (cl-return-from persp-state-save))
-  (let ((target-file (if (and file (not (string-equal "" file)))
+  (let ((target-file (if (and (bound-and-true-p file) (not (string-equal "" file)))
```

Comment:
I think this is the reason why this patch breaks the test suite. This is not correct use of `bound-and-true-p`: it will return `nil` when called interactively because `file` is in lexical scope. The documentation for `boundp` states:

> Note that if ‘lexical-binding’ is in effect, this refers to the
global value outside of any lexical scope.

We now use `lexical-binding` in Perspective.